sbin/cubeit

```diff
@@ -173,6 +178,94 @@ if [ -n "$ARTIFACTS_DIR" ]; then
 fi
 export ARTIFACTS_DIR
 
+# determine the location of key store
+if [ -z "$KEYS_DIR" -o ! -d "$KEYS_DIR" ]; then
+    SAMPLE_KEYS_DIR="$ARTIFACTS_DIR/sample-keys"
+    USER_KEYS_DIR="$ARTIFACTS_DIR/user-keys"
+
+    if [ -d "$USER_KEYS_DIR" ]; then
+        KEYS_DIR="$USER_KEYS_DIR"
+    else
+        KEYS_DIR="$SAMPLE_KEYS_DIR"
+    fi
+fi
+
+# check if the filesystem can support xattr.
+check_fs_for_ima()
+{
+    local tmpfile="`mktemp test_XXXX`"
+    local res=`setfattr -x security.ima "$tmpfile" 2>&1 | grep -q 'Operation not supported$'`
+
+    rm -f "$tmpfile"
+
+    return ! $res
+}
+
+if [ $DO_IMA_SIGN -eq 1 ]; then
+    if [ ! -s "$KEYS_DIR/ima_keys/ima_privkey.pem" ]; then
+        debugmsg ${DEBUG_CRIT} "[ERROR] Unable to run IMA signing due to the invalid signing key"
+        exit 1
+    elif ! check_fs_for_ima; then
+        debugmsg ${DEBUG_CRIT} "[ERROR]: Current file system does not support to set xattr"
+        exit 1
+    fi
+fi
+
+ima_sign()
+{
+    evmctl ima_sign --hashalgo sha256 --rsa \
+        --key "$KEYS_DIR/ima_keys/ima_privkey.pem" \
+        -r -t f "$1" && {
+        return 0
+    } || {
+        debugmsg ${DEBUG_INFO} "[ERROR]: IMA signing occurs with error for $1."
+        exit 1
+    } &
+    pidspinner "$!" "1"
+}
+
+repack_image_for_ima()
+{
+    local tarball="$1"
+    local tarball_dst=$2
+    local repack_dir=`mktemp -d repack_tarball.XXXXXX`
+    local repack_image="`basename ${tarball}`"
```

Comment:
`fooo` is no longer recommended. The installer code is a mix of old and new, any new subshell commands are all using $(foo), which is the recommended format.